CHANGELOG.md

```diff
@@ -46,6 +56,13 @@ Changes prior to 3.9.0 are documented as [release notes on GitHub](https://githu
       return instance;
     }
     ```
+- These following entities will be renamed (and removed) in an upcoming major release. To support incremental migration, both old and new names are still provided.
+  - `bsoncxx::type` -> `bsoncxx::types::id`.
```

Comment:
Consider adding `BSONCXX_DEPRECATED` on the deprecated symbols. I expect that might help inform users with compiler warnings. Though I expect it will certainly trigger (possibly many) deprecation warnings for users.